pulp_ansible/app/tasks/signature.py

```diff
@@ -71,13 +73,20 @@ async def sign_collection_version(self, collection_version):
         # Limits the number of subprocesses spawned/running at one time
         async with self.semaphore:
             # We use the manifest to create the signature
-            async with aiofiles.tempfile.TemporaryDirectory(dir=".") as d:
-                # OpenPGP doesn't take filename into account for signatures, not sure about others
-                async with aiofiles.open(f"{d}/MANIFEST.json", "w") as t:
-                    await t.write(json.dumps(collection_version.manifest))
-                result = await self.signing_service.asign(t.name)
-                async with aiofiles.open(result["signature"], "rb") as sig:
-                    data = await sig.read()
+            # OpenPGP doesn't take filename into account for signatures, not sure about others
+            async with aiofiles.tempfile.NamedTemporaryFile(dir=".", mode="wb") as m:
+                cartifact = await sync_to_async(
+                    collection_version.contentartifact_set.select_related("artifact").first
+                )()
```

Comment:
Not sure if there is a better way to get this content_artifact to extract the artifact.

Reply:
No, this looks right to me.